Makefile

```diff
@@ -20,6 +21,9 @@ protoc_go_package    := github.com/golang/protobuf/protoc-gen-go
 thrift_gen_package   := github.com/uber/tchannel-go
 thrift_output_dir    := generated/thrift/rpc
 thrift_rules_dir     := generated/thrift
+m3x_package          := github.com/m3db/m3x
+m3x_package_path     := $(gopath_prefix)/$(m3x_package)
```

Comment:
Would it be better to rely on the vendor'd version in m3db?

Reply:
Hm main concern is if someone has a dirty version of `m3x` in their workspace. Could you add checks to fail if m3x is not on `master`

Reply:
Sure thing, added check.